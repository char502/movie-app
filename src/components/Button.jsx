import React from 'react';
import styled, {css} from 'styled-components';
import {rem} from 'polished'

// const StyledButton = styled.button`
//   font-family: impact;
//   font-size: ${(props) => props.theme.fonts.medium};
//   border: 0;
//   border-radius: ${(props) => props.theme.borderRadius};
//   margin-right: ${rem(8)};
//   background: ${(props) => props.theme.gradient};
//   color: white;
// `;

const StyledButton = styled.button`
    ${(props) => {
        const {theme} = props;
        return css`
            font-family: impact;
            font-size: ${theme.fonts.medium};
            border: 0;
            border-radius: ${theme.borderRadius};
            margin-right: ${rem(8)};
            background: ${theme.gradient};
            color: white;
        `
    }}
`

export const Button = (props) => <StyledButton onClick={props.onClick}>{props.label}</StyledButton>