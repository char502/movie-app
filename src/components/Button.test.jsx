import React from "react";
import {shallow} from 'enzyme';
import { Button , MediaPageButton } from './Button';
import toJson from 'enzyme-to-json';

it('Button Snapshots', () => {
    const buttonWrap = shallow(<Button onClick={() => {}} label={"TEST"} />)
    expect(toJson(buttonWrap)).toMatchSnapshot();
    const mediaPageButtonWrap = shallow(<MediaPageButton />)
    expect(toJson(mediaPageButtonWrap)).toMatchSnapshot();
});