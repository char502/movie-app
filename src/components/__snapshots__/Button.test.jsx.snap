// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshots 1`] = `
<styled.button
  onClick={[Function]}
>
  TEST
</styled.button>
`;

exports[`Button Snapshots 2`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  font-size: ",
          [Function],
          ";
  color: black;
  background: #e8e8e8;
  font-weight: 800;
  padding: ",
          [Function],
          "
    ",
          [Function],
          ";
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: ",
          [Function],
          ";
  border: 1px solid #e8e8e8;
  cursor: pointer;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.button",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "button",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
